### DISCUSSION TITLE
Button to switch audio files

### DISCUSSION THREAD

**Discussion by Elgoritm** [katspaugh/wavesurfer.js#2846] [Q&A]
Hey there, 

If I remember correctly, in the pre-alpha / beta I was able to switch audio like this: 

the wave surfer is first created with some options
then 

```
const A = '';
const B = '';
const audioUrl = [A, B];

document.querySelector('.button').addEventListener('click', () => {
audioUrl.reverse();
ws.load(`${audioUrl[0]}`);  
})

ws.load(`${audioUrl[0]}`);  
```

It would reload and keep the cursor in the same spot it was in at the current playback .
When I do this now, the cursor moves back to the start of the file and completely breaks the player to being super slow. I'm probably doing something silly.. Any help would be appreciated:)

**Comment by katspaugh** [comment]
That it resets it back to 0 is expected. Shouldn’t be slow though. Does it happen for you in the React example on wavesurfer-js.org?

**Reply by Elgoritm** [reply]
Hm I wonder what I did that made it switch the cursor in place before.. It is the behaviour I'd like to have. 

The React example works fine in the browser. I think maybe part of the problem is that I'm using some other libraries that work with audioContext / sourceNodes to hook things up. 

I think if I'm able to update the wave surfer options.media with the new audio it would work? I am able to update it, just not able to get wave surfer to update after the options update..

**Reply by katspaugh** [reply]
I would just remember the current time, call load with a new URL and then setTime on ready to set it to the old position.
